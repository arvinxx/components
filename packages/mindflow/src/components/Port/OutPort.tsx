import React from 'react';
import type { FC } from 'react';

import './style.less';

import CollapseIcon from '../CollapseIcon';

interface OutPortProps {
  color: string;
  collapsed: boolean;
  visible: boolean;
  id: string;
}

export const OutPort: FC<OutPortProps> = ({
  color,
  collapsed,
  visible,
  id,
}) => {
  return !visible ? null : (
    <div className="mind-port-out-port">
      <CollapseIcon id={id} color={color} collapsed={collapsed} />
    </div>
  );
};
