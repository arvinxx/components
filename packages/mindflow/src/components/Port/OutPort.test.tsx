import React from 'react';
import { render } from '@testing-library/react';

import { OutPort } from './OutPort';

describe('OutPort', () => {
  it('默认', () => {
    const { container } = render(
      <OutPort collapsed={false} color={'red'} id={'123'} />,
    );
    expect(container).toMatchSnapshot();
  });
});
