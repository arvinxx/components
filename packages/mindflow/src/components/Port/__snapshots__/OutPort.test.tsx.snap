// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutPort 默认 1`] = `
<div>
  <div
    class="mind-port-out-port"
  >
    <div
      class="mind-node-collapse"
      data-testid="icon"
      style="border-color: red; background: white;"
    >
      <div
        class="mind-node-collapse-icon"
        style="color: red;"
      >
        <span
          aria-label="minus"
          class="anticon anticon-minus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="minus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
